internal/service/alarms/api/server.go

```diff
@@ -492,14 +492,41 @@ func (a *AlarmsServer) UpdateAlarmServiceConfiguration(ctx context.Context, requ
 
 }
 
-func (a *AlarmsServer) GetProbableCauses(ctx context.Context, request api.GetProbableCausesRequestObject) (api.GetProbableCausesResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+// GetProbableCauses list all the relevant probableCauses based on alarms that are currently in the events table
+func (a *AlarmsServer) GetProbableCauses(ctx context.Context, _ api.GetProbableCausesRequestObject) (api.GetProbableCausesResponseObject, error) {
+	defs, err := a.AlarmsRepository.GetAllAlarmDefForProbableCause(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get alarm definitions for probableCauses: %w", err)
+	}
+
+	pcs := make([]api.ProbableCause, 0, len(defs))
+	for _, def := range defs {
+		pcs = append(pcs, models.ConvertDefToPC(def))
+	}
+
+	slog.Info("Successfully retrieved probableCauses of current events", "count", len(pcs))
+	return api.GetProbableCauses200JSONResponse(pcs), nil
 }
 
+// GetProbableCause retrieve a probable cause of alarm with UUID
 func (a *AlarmsServer) GetProbableCause(ctx context.Context, request api.GetProbableCauseRequestObject) (api.GetProbableCauseResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+	alarmDefs, err := a.AlarmsRepository.GetAlarmDefForProbableCause(ctx, request.ProbableCauseId)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get alarm definition for probableCause: %w", err)
+	}
+
+	if len(alarmDefs) == 0 {
+		return api.GetProbableCause404ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+			AdditionalAttributes: &map[string]string{
+				"probableCauseID": request.ProbableCauseId.String(),
+			},
+			Detail: "could not find the probable cause for given probableCauseID",
```

Comment:
Would it be better to have the `request.ProbableCauseId` also included here? I'm thinking about situations where multiple requests come where some fail and some are successful, it would be easier to just include which failed and which not. Please disregard if it doesn't apply to the alarms usecase.

Reply:
The `request.ProbableCauseId` is included in the AdditionalAttributes just above.... is that what you meant?